gradio/upload.py

```diff
@@ -0,0 +1,173 @@
+        if cell["cell_type"] == "code":
+            for line in cell["source"]:
+                spaces = ""
+                if line.startswith(" ") or line.startswith("\t"):
+                    spaces = line[: len(line) - len(line.lstrip())]
+                    line = line.strip()
+                if line.startswith("!pip install") or line.startswith("%pip install"):
+                    dependencies.extend(line[13:].strip().split(" "))
+                elif line.startswith("%"):
+                    continue
+                elif line.startswith("!"):
+                    command = line[1:].rstrip()
+                    subprocess_call = f"subprocess.run('{command}', shell=True, check=True, text=True)"
+                    script_content.append(spaces + subprocess_call)
+                    import_subprocess = True
+                else:
+                    script_content.append(spaces + line)
+
+            script_content.append("\n")
+
+    if import_subprocess:
+        script_content.insert(0, "import subprocess\n")
+
+    script = "".join(script_content)
+    with open(script_path, "w", encoding="utf-8") as f:
+        f.write(script)
+    return dependencies
+
+
+def upload_to_spaces():
+    if os.getenv("SYSTEM") == "spaces":  # in case a notebook has an upload call
+        return
+
+    hf_api = huggingface_hub.HfApi()
+    whoami = None
+    login = False
+    try:
+        whoami = hf_api.whoami()
+        if whoami["auth"]["accessToken"]["role"] != "write":
+            login = True
+    except:
+        login = True
+    if login:
+        print("Need 'write' access token to create a Spaces repo.")
+        huggingface_hub.login(add_to_git_credential=False)
+        whoami = hf_api.whoami()
+    repo_directory = os.getcwd()
+    readme_file = os.path.join(repo_directory, "README.md")
+    if not os.path.exists(readme_file):
+        with open(readme_file, "w") as f:
+            f.write("")
+    configuration = get_existing_configuration(readme_file)
+    if not configuration:
+        print(f"Creating new Spaces Repo in {repo_directory}")
+        add_configuration_to_readme(repo_directory, readme_file)
+        configuration = get_existing_configuration(readme_file)
+    requirements_file = os.path.join(repo_directory, "requirements.txt")
+    dependencies = []
+    if configuration.get("notebook_file"):
+        dependencies = notebook_to_script(
+            configuration["notebook_file"], configuration["app_file"]
+        )
+    if (
+        not os.path.exists(requirements_file)
+        and input("Create requirements.txt file? [y/n]: ").lower() == "y"
+    ):
+        with open(requirements_file, "w") as f:
+            f.write("\n".join(dependencies))
+        input(
+            "Created requirements.txt file. Please add any dependencies to this file. Press enter to continue. "
+        )
+    name = whoami["name"]
+    space_id = f"{name}/{configuration['title']}"
+
+    huggingface_hub.create_repo(
```

Comment:
`create_repo` returns a RepoUrl object with a repo_id attribute. Instead of doing "whoami" to get the namespace, it can be implicitly inferred by the server and returned.

So instead of:
```py
space_id = f"{name}/{configuration['title']}"
huggingface_hub.create_repo(space_id, ...)
```

you should do
```py

space_id = huggingface_hub.create_repo(configuration["xxx"], ...).repo_id
```

That way, user can be prompted to chose the repo_id with or without namespace (e.g. either `dolly-v2` => implicit `username/dolly-vy` or `my-organization/dolly-v2` => explicit org)